tests/unit/test_extractor.py

```diff
@@ -0,0 +1,125 @@
+from cdf_fabric_replicator.extractor import CdfFabricExtractor
+from cognite.extractorutils.base import CancellationToken
+from cognite.extractorutils.statestore import LocalStateStore
+
+
+@pytest.fixture(scope="session")
+def extractor(config):
+    stop_event = CancellationToken()
+    extractor = CdfFabricExtractor(stop_event=stop_event)
+    extractor.config = config
+    extractor.client = extractor.config.cognite.get_cognite_client("test_extractor")
+    extractor.state_store = LocalStateStore(extractor.config.extractor.state_store.local.path)
+    
+    yield extractor
+
+
+def test_parse_abfss_url(extractor):
+    url = 'https://container@account.dfs.core.windows.net/path'
+    result = extractor.parse_abfss_url(url)
+    assert result == ('container', 'account', '/path')
+
+
+def test_write_time_series_to_cdf(extractor, mocker):
+    data = {
+        "externalId": ["id1", "id2", "id1", "id2"],
+        "timestamp": [
+            "2022-02-07T16:01:27.001Z",
+            "2022-02-07T16:03:47.000Z",
+            "2022-02-07T16:03:52.000Z",
+            "2022-02-07T16:04:12.000Z",
+        ],
+        "value": [1.0, 2.0, 3.0, 4.0],
+    }
+    df = pd.DataFrame(data)
+
+    mocker.patch.object(extractor.state_store, 'get_state', return_value=(None, ))
+    mocker.patch.object(extractor.client.time_series.data, 'insert_dataframe', return_value=None)
+    mocker.patch("cdf_fabric_replicator.extractor.CdfFabricExtractor.set_state", return_value=None)
+
+    extractor.write_time_series_to_cdf(df)
+
+    extractor.state_store.get_state.assert_any_call('/table/path-id1-state')
+    extractor.state_store.get_state.assert_any_call('/table/path-id2-state')
+    assert(extractor.client.time_series.data.insert_dataframe.call_count == 2)
+    extractor.set_state.assert_any_call("/table/path-id1-state", df[df["externalId"] == "id1"]["timestamp"].max())
+    extractor.set_state.assert_any_call("/table/path-id2-state", df[df["externalId"] == "id2"]["timestamp"].max())
+
+
+def test_write_time_series_to_cdf_filter_old_data_points(extractor, mocker):
+    data = {
+        "externalId": ["id1", "id1", "id1", "id1"],
+        "timestamp": [
+            "2022-02-07T16:01:27.001Z",
+            "2022-02-07T16:03:47.000Z",
+            "2022-02-07T16:03:52.000Z",
+            "2022-02-07T16:04:12.000Z",
+        ],
+        "value": [1.0, 2.0, 3.0, 4.0],
+    }
+    df = pd.DataFrame(data)
+
+    last_update_time = "2022-02-07T16:03:47.000Z"
+
+    mocker.patch.object(extractor.state_store, "get_state", return_value=(last_update_time,))
+    mocker.patch.object(extractor.client.time_series.data, "insert_dataframe", return_value=None)
+    mocker.patch("cdf_fabric_replicator.extractor.CdfFabricExtractor.set_state", return_value=None)
+
+    extractor.write_time_series_to_cdf(df)
+
+    expected_update_list = df[df["timestamp"] > last_update_time]
+
+    extractor.state_store.get_state.assert_any_call("/table/path-id1-state")
```

Comment:
same as above ... should those be consts or config?

Reply:
I can change these to be configurable checks that would probably be the best option

Reply:
I actually found a way for me to do this where I can take those unique ids and iterate over them to ensure calls for both of the assert call methods like this:
 
 ```python
 for external_id in list(set(data["externalId"])):
        extractor.state_store.get_state.assert_any_call(f'/table/path-{external_id}-state')
        extractor.set_state.assert_any_call(f"/table/path-{external_id}-state", df[df["externalId"] == external_id]["timestamp"].max())
```

and I just updated those to this as well